packages/core/src/RushTool.ts

```diff
@@ -0,0 +1,64 @@
+import path from "path";
+import fs from "fs-extra";
+import jju from "jju";
+
+import { PackageJSON } from "@changesets/types";
+import { Tool, ToolType, Package, Packages, MonorepoRoot } from './Tool';
+
+interface RushJson {
+    projects: RushProject[];
+}
+
+interface RushProject {
+    packageName: string;
+    projectFolder: string;
+}
+
+export const RushTool: Tool = {
+    type(): ToolType {
+        return 'rush';
+    },
+
+    async isMonorepoRoot(directory: string): Promise<MonorepoRoot | undefined> {
+        try {
+            await fs.readFile(path.join(directory, "rush.json"), "utf8");
+            return {
+                dir: directory,
+                tool: RushTool
+            };
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+    },
+
+    async getPackages(directory: string): Promise<Packages | undefined> {
+        try {
+            const rushText: string = await fs.readFile(path.join(directory, "rush.json"), "utf8");
+
+            // Rush configuration files are full of inline and block-scope comment blocks (JSONC),
+            // so we use a parser that can handle that.
+            const rushJson: RushJson = jju.parse(rushText);
+
+            const directories = rushJson.projects.map(project => path.resolve(directory, project.projectFolder)).sort();
```

Comment:
It's quite interesting that you `.sort()` this here. Is this strictly necessary?

Reply:
🤔 I guess there's a couple options there. I think the answer is _yes_, we do need to sort it, because Rush's own tools (like `rush list`) sorts by package name, and so does the existing `changesets` ascii UI, so users would expect that.

But, do we need to sort it _here_? I think the answer is yes, because I think it's useful if you're implementing a Tool to assume that the list of packages you return will be displayed essentially as-is to users, and not sorted later by get-packages or changesets. This gives the Tool author some flexibility on exactly what "sorting" means -- here it's clearly alphabetical (although perhaps I should make that explicit), whereas if I imagine a later cross-language Tool like a `SwiftProjectYamlTool` or a `AndroidBuildGradleTool`, perhaps an expected ordering of packages is very different.

Reply:
I would probably be very wary of trusting the input to be sorted anyhow. That being said, I don't mind sorting here but I'm also not sure when that "trait" would actually be helpful. This should only matter for display purposes and atm Changesets only display packages by their name (and here we are sorting by the FS path).

Reply:
Good points! I will strip sorting from any Tools that currently do it, and the end consuming UI will do the sorting instead.